### PR TITLE
vmalert: add `-rule.evalDelay` flag and `eval_delay` as group attribute

### PR DESCRIPTION
**Pull request by Haleygo**
address https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5155.

And as described in above issue
>there are three ways to modify the timestamp of raw series queried from database:
>
> 1. set -search.latencyOffset on vmselect;
> 2. set -datasource.lookback on vmalert;
> 3. add latency_offset in group's params, which will be attached to group's evaluation query requests and override -search.latencyOffset if there is.

`-datasource.lookback` used perform the same as the other two options(both of them have effect on datasource side) by modifying the `time` parameter in evaluation requests, and that will break what `eval_delay` trying to fix here.
So I think we can also mark `-datasource.lookback` as "will be deprecated" here, and recommend users to use the other two options in the future which should cover the needs.

Please note, `eval_delay` won't work if there is `eval_offset` in group.

### CODE REVIEW

File: app/vmalert/README.md [hunk 1]
```diff
@@ -119,6 +119,12 @@ name: <string>
 # `eval_offset` can't be bigger than `interval`.
 [ eval_offset: <duration> ]
 
+# Optional
+# Adjust the `time` parameter of group evaluation requests to match intentional query delay from datasource.
```
Comment (hagen1778) [inline app/vmalert/README.md:123]: I think we should do better explaining this. Remember, user should receive enough of understanding from this decription to decide whether to change this param.

File: app/vmalert/README.md [hunk 2]
```diff
@@ -119,6 +119,12 @@ name: <string>
...
+# This will override global `-rule.evalDelay` flag if both be using.
+# See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5155.
+[ eval_delay: <duration> ]
```
Comment (hagen1778) [inline app/vmalert/README.md:126]: Should we have a default value here? For example, 30s default delay will compensate default latency offset

File: app/vmalert/README.md [hunk 3]
```diff
@@ -805,8 +809,9 @@ at least two times bigger than the resolution.
 > Please note, data delay is inevitable in distributed systems. And it is better to account for it instead of ignoring.
 
 By default, recently written samples to VictoriaMetrics aren't visible for queries for up to 30s
-(see `-search.latencyOffset` command-line flag at vmselect). Such delay is needed to eliminate risk of incomplete
+(see `-search.latencyOffset` command-line flag at vmselect, and it can be overridden by adding `latency_offset` to group's params). Such delay is needed to eliminate risk of incomplete
 data on the moment of querying, since metrics collectors won't be able to deliver the data in time.
+If you see delayed evaluation results caused by this intentional query delay from datasource, you can set `-rule.evalDelay` equal with `-search.latencyOffset` to fix the gap.
```
Comment (hagen1778) [inline app/vmalert/README.md:814]:
```suggestion
If you observe a delay in timestamps for produced evaluation results caused by this intentional query delay from datasource, try setting `-rule.evalDelay` equal to `-search.latencyOffset` (a cmd-line flag configured for VictoriaMetrics single-node or vmselect) to compensate the delay.
```

File: app/vmalert/datasource/init.go [hunk 4]
```diff
@@ -85,6 +86,9 @@ func Init(extraParams url.Values) (QuerierBuilder, error) {
 	if !*queryTimeAlignment {
 		logger.Warnf("flag `datasource.queryTimeAlignment` is deprecated and will be removed in next releases, please use `eval_alignment` in rule group instead")
 	}
+	if *lookBack != 0 {
+		logger.Warnf("flag `datasource.lookback` will be deprecated soon, please use `-search.latencyOffset` in datasource or override it by adding `latency_offset` in rule group's params.")
```
Comment (hagen1778) [inline app/vmalert/datasource/init.go:90]:
```suggestion
		logger.Warnf("flag `datasource.lookback` will be deprecated soon, please adjust `-search.latencyOffset`  at datasource side or specify `latency_offset` in rule group's params.")
```

Comment (hagen1778) [reply app/vmalert/datasource/init.go:90]: please link ticket here as well

File: docs/CHANGELOG.md [hunk 5]
```diff
@@ -37,6 +38,7 @@ The sandbox cluster installation is running under the constant load generated by
 * FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert.html): add `eval_alignment` attribute for [Groups](https://docs.victoriametrics.com/vmalert.html#groups), it will align group query requests timestamp with interval like `datasource.queryTimeAlignment` did.
   This also means that `datasource.queryTimeAlignment` command-line flag becomes deprecated now and will have no effect if configured. If `datasource.queryTimeAlignment` was set to `false` before, then `eval_alignment` has to be set to `false` explicitly under group.
   See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5049).
+* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert.html): add `-rule.evalDelay` flag and `eval_delay` attribute for [Groups](https://docs.victoriametrics.com/vmalert.html#groups), it will adjust the `time` parameter of rule evaluation requests to match intentional query delay from datasource. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5155).
```
Comment (hagen1778) [inline docs/CHANGELOG.md:41]:
```suggestion
* FEATURE: [vmalert](https://docs.victoriametrics.com/vmalert.html): add `-rule.evalDelay` flag and `eval_delay` attribute for [Groups](https://docs.victoriametrics.com/vmalert.html#groups). The new flag and param can be used to adjust the `time` parameter for rule evaluation requests to match intentional query delay from the datasource. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5155).
```

File: app/vmalert/README.md (hunk 1 above)
Comment (Haleygo) [reply app/vmalert/README.md]: Updated, plz check.

File: app/vmalert/README.md (hunk 2 above)
Comment (Haleygo) [reply app/vmalert/README.md:126]: Good idea, set to 30s now.

File: app/vmalert/datasource/init.go (hunk 4 above)
Comment (Haleygo) [reply app/vmalert/datasource/init.go]: Sure!

**Review by hagen1778 — APPROVED** [review]
LGTM!

**Pull request merged**